flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/netty/NettyConnectionWriterImpl.java

```diff
@@ -30,24 +30,33 @@ public class NettyConnectionWriterImpl implements NettyConnectionWriter {
 
     private final NettyConnectionId connectionId;
 
-    private final BufferAvailabilityListener availabilityListener;
+    private Runnable availabilityListener;
 
+    public NettyConnectionWriterImpl(NettyPayloadManager nettyPayloadManager) {
+        this.nettyPayloadManager = nettyPayloadManager;
+        this.connectionId = NettyConnectionId.newId();
+    }
+
+    @VisibleForTesting
```

Comment:
Do we need this `VisibleForTesting`? Could we test the logic by the existing public/protected methods?